codelab/02_kpt-deploy/tutorial.md

```diff
@@ -0,0 +1,240 @@
+*  You can prune your resources accurately with [a three-way merge strategy](https://kubectl.docs.kubernetes.io/pages/app_management/field_merge_semantics.html). 
+
+#### What you'll learn in this codelab
+    
+*   The difference between `kpt` pruning and `kubectl` pruning
+*   How to enable kpt to deploy your configurations in their live state.
+*   How to use kpt with kustomize.
+
+This codelab is the second session. Check out the [01_kpt-validate](https://github.com/GoogleContainerTools/skaffold/tree/master/codelab/01_kpt-validate) about how to use kpt to validate your configuration before deploying to the cluster.
+
+## Prerequisites
+
+If you are new to skaffold, you can check out [the skaffold tutorials](https://skaffold.dev/docs/tutorials/) to get a basic idea. 
+Or just follow this codelab, we will explain what happens in each step.
+
+#### Install `skaffold`
+
+*   Check if `skaffold` is installed and the installed version is >= v1.17.0
+    ```bash
+    skaffold version
+    ```
+
+*   If you haven't installed `skaffold` previously, run
+    ```bash
+    curl -Lo skaffold https://storage.googleapis.com/skaffold/builds/latest/skaffold-linux-amd64 && \
+    sudo install skaffold /usr/local/bin/ | bash
+    ```  
+*   To upgrade skaffold to a newer version, run
+    ```bash
+    sudo rm -f /usr/local/bin/skaffold && curl -Lo skaffold https://storage.googleapis.com/skaffold/builds/latest/skaffold-linux-amd64 && \
+    sudo install skaffold /usr/local/bin/ | bash
+    ```
+    
+#### Install `kpt`
+
+*   Check if `kpt` is installed and the installed version is >= 0.34.0
+    ```bash
+    kpt version
+    ```
+*   If you haven't installed `kpt` previously, run
+    ```bash
+    sudo apt-get install google-cloud-sdk-kpt
+    ```  
+    
+#### Install `kustomize`
+
+*   Check if `kustomize` is installed and the installed version is >= v3.4.3
+    ```bash
+    kustomize version
+    ```
+*   If you haven't installed `kustomize` previously, run
+    ```bash
+    curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash &&  sudo mv kustomize /usr/local/go/bin
+    ```  
+    
+## Getting started
+
+Time to complete: **About 3 minutes**
+
+
+*   Run this command to start your local minikube cluster:
+    
+    ```bash
+    minikube start
+    ```
+    
+*   Once your cluster is set up, the following message is displayed:
+    
+    ```terminal
+    Done! kubectl is now configured to use "minikube"
+    ```
+    
+*   Now let's use `kpt pkg` to download the application example. 
+        
+    ```bash
+    kpt pkg get https://github.com/GoogleContainerTools/skaffold.git/codelab/02_kpt-deploy/resources/sample-app guestbook-cl2 && cd guestbook-cl2
+    ```
+
+<walkthrough-pin-section-icon></walkthrough-pin-section-icon>
+**Tips**
+>  If you haven't used `skaffold dev`, you can go through this [codelab](https://github.com/GoogleContainerTools/skaffold/tree/master/codelab/01_kpt-validate).
```

Comment:
this tip seems a little bit out of place in this section. Is there supposed to be information about `skaffold dev` here?

Reply:
yeah, the first codelab gives a brief intro about `skaffold dev` as well as running it against a real example.